src/org/labkey/targetedms/outliers/OutlierGenerator.java

```diff
@@ -312,18 +318,28 @@ private void retainIfPreferred(Map<String, QCMetricConfiguration> preferredConfi
      * set of results.
      */
     @NotNull
-    private Map<Long, RawMetricDataSet.PrecursorInfo> loadPrecursors(TargetedMSSchema schema) throws SQLException
+    private Map<Long, RawMetricDataSet.PrecursorInfo> loadPrecursors(TargetedMSSchema schema, Map<Long, Object> excludedPrecursorsIds, boolean showExcludedPrecursors) throws SQLException
     {
         Map<Long, RawMetricDataSet.PrecursorInfo> precursors = new HashMap<>();
 
         DecimalFormat format = new DecimalFormat();
         format.setMinimumFractionDigits(4);
 
+        Collection<Map<String, Object>> excludedPrecursors = new TableSelector(schema.getTable("ExcludedPrecursors")).getMapCollection();
+
         // First the proteomics side
         try (ResultSet rs = new TableSelector(schema.getTable(TargetedMSSchema.TABLE_PRECURSOR)).getResultSet(false))
         {
             while (rs.next())
             {
+                if (!showExcludedPrecursors && isExcludedPrecursorPeptide(excludedPrecursors,
+                        rs.getString("ModifiedSequence"),
+                        rs.getInt("Charge"),
+                        rs.getDouble("MZ")))
+                {
+                    excludedPrecursorsIds.put(rs.getLong("Id"), null);
```

Comment:
Why is excludedPrecursorIds a map? If the value is always null, please use a Set.